docs/src/tutorials/conversions_views.md

```diff
@@ -507,3 +507,33 @@ Base.ReshapedArray{RGB{Normed{UInt8,8}},3,Base.ReinterpretArray{RGB{Normed{UInt8
 While there is little or no performance cost to making use of
 JuliaImage's convenient views, sometimes the types can get
 complicated!
+
+## Preserving colorant type while conversion
+
+While using Float32, Float64 to convert the raw storage type of an image 
+the color space information is lost. Replacaing Float32, Float64 with their lower-case
+forms preserves the colorant type.
+
+Take an example of converting a Gray image to Float64. Simply broadcating Float64
+will promote the storage to Float64 but the color space information is lost.
+However, when we use float64. The colorant type inforamtaion is preserved.
```

Comment:
Another major difference between `Float64` and `float64` is that `Float64.(rgb_img)` will error while the `float64.(rgb_img)` one works as expected.

Reply:
why does `Float64.(rgb_img)` throws error?

Reply:
```julia
julia> using Colors

julia> c = Gray(0.5)
Gray{Float64}(0.5)

julia> Float64(c)
0.5

julia> c = RGB(0.2, 0.4, 0.8)
RGB{Float64}(0.2,0.4,0.8)

julia> Float64(c)
ERROR: MethodError: no method matching Float64(::RGB{Float64})
Closest candidates are:
  Float64(::BigFloat) at mpfr.jl:332
  Float64(::BigFloat, ::Base.MPFR.MPFRRoundingMode) at mpfr.jl:332
  Float64(::BigFloat, ::RoundingMode) at mpfr.jl:334
  ...
Stacktrace:
 [1] top-level scope at REPL[5]:1
```
Tell me what `Float64` value should `Float64(RGB(0.2, 0.4, 0.8))` return? There are 3 color channels.

My comment below points out it's a little more complicated because there is an agreed-upon standard for converting RGB to a pure grayscale intensity. So
```julia
julia> Float64(Gray(c))
0.38580000400543213
```
The question is whether `Float64(c)` should do that on its own. The concern is that too many people will want to to mean element-wise conversion, which you instead should do this way:
```julia
julia> RGB{Float32}(c)
RGB{Float32}(0.2f0,0.4f0,0.8f0)
```
or in shorthand
```julia
julia> using ImageCore

julia> float32(c)
RGB{Float32}(0.2f0,0.4f0,0.8f0)
```

JuliaImages tries not to read your mind. We ask programmers to be clear about what they want, otherwise subtle bugs are more likely and one always loses a lot of time to tracking them down.

Reply:
Thanks for explaining.

Reply:
Glad it helped. And to be sure that the more general principle is also clear: if `T` is a type, `T(x)` should return an object of type `T`. Since `Float64` is a type, it needs to return a single `Float64` object. But `float64` is not a type (it's a function), so it's under no such constraints. So we made it operate on the "storage type" without modifying the "color space." That's what I meant below about "orthogonality," essentially keeping the two concepts separate from one another.